var/spack/packages/R/package.py

```diff
@@ -1,33 +1,49 @@
 from spack import *
 
+
 class R(Package):
-    """R is 'GNU S', a freely available language and environment for
-       statistical computing and graphics which provides a wide va
-       riety of statistical and graphical techniques: linear and
-       nonlinear modelling, statistical tests, time series analysis,
-       classification, clustering, etc. Please consult the R project
-       homepage for further information."""
-    homepage = "http://www.example.com"
-    url      = "http://cran.cnr.berkeley.edu/src/base/R-3/R-3.1.2.tar.gz"
+    """
+    R is 'GNU S', a freely available language and environment for statistical computing and graphics which provides a
+    wide variety of statistical and graphical techniques: linear and nonlinear modelling, statistical tests, time series
+    analysis, classification, clustering, etc. Please consult the R project homepage for further information.
+    """
+    homepage = "https://www.r-project.org"
+    url = "http://cran.cnr.berkeley.edu/src/base/R-3/R-3.1.2.tar.gz"
 
+    version('3.2.3', '1ba3dac113efab69e706902810cc2970')
+    version('3.2.2', '57cef5c2e210a5454da1979562a10e5b')
+    version('3.2.1', 'c2aac8b40f84e08e7f8c9068de9239a3')
+    version('3.2.0', '66fa17ad457d7e618191aa0f52fc402e')
+    version('3.1.3', '53a85b884925aa6b5811dfc361d73fc4')
     version('3.1.2', '3af29ec06704cbd08d4ba8d69250ae74')
 
-    depends_on("readline")
-    depends_on("ncurses")
-    depends_on("icu")
-    depends_on("glib")
-    depends_on("zlib")
-    depends_on("libtiff")
-    depends_on("jpeg")
-    depends_on("cairo")
-    depends_on("pango")
-    depends_on("freetype")
-    depends_on("tcl")
-    depends_on("tk")
+    variant('external-lapack', default=False, description='Links to externally installed BLAS/LAPACK')
+
+    # Virtual dependencies
+    depends_on('blas', when='+external-lapack')
+    depends_on('lapack', when='+external-lapack')
```

Comment:
Sorry, I've gotten behind on the PR's.  I think that merging as-is is fine right now.  I'll merge it.

I think we should have a separate discussion on BLAS and LAPACK though, as I'm not convinced that combining them is the best thing... OpenBLAS actually installs its own netlib LAPACK and I think it would be nice if that weren't duplicated.  While it is not the largest package in the world, it would be nice to guarantee the user at least somewhat that if they install a particular BLAS and lapack, they'll link against them.

I think two things could help with this.  First, fixing #311 would make it much easier to build up a consistent stack like this.  You could install the BLAS and LAPACK you want, and things would then just use them.  Second, I would need something to pass the `+shared` or `~shared` variants through virtual deps, so that you could depend on `blas+shared`.  Right now I think the lack of that capability complicates `R`.  Third, once we have concretization preferences in Spack, it might be easier to make sane initial choices for combinations of packages.  That would let you specify a sane default BLAS and LAPACK for cases where the user didn't.

Since none of that is done yet, I'll just merge this.  But I'd be interested to hear what you think.  I didn't realize when I merged OpenBLAS that it actually just installed its own `netlib-lapack`.  As I said, I'd eventually I'd like to have enough controls on the concretization process to allow the flexibility of separate BLAS and LAPACK implementations, while still having a sane default stack (or several).  Does that make sense/sound good?